### ISSUE TITLE
Chain call order

### ISSUE THREAD

**Issue by bryanjhv** [fastify/fluent-json-schema#31]
Let's take this simple example:

```js
const S = require('fluent-schema')

console.log(
  S.object()
    .prop(
      'name',
      S.string()
        .required()
        .minLength(3)
        .maxLength(35)
    )
    .valueOf()
)
```

From what I understand, it should print the schema on the command line. What it currently does is:

```txt
/tmp/node_modules/fluent-schema/src/utils.js:154
  return options.factory({ schema: { ...schema, [key]: value }, ...options })
                 ^

TypeError: options.factory is not a function
    at setAttribute (/tmp/node_modules/fluent-schema/src/utils.js:154:18)
    at Object.minLength (/tmp/node_modules/fluent-schema/src/StringSchema.js:49:12)
    at Object.<anonymous> (/tmp/test.js:9:10)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:742:12)
    at startup (internal/bootstrap/node.js:283:19)
```

However, if I move the `.required()` call to the end of the prop definition, it works as expected.

**Comment by aboutlo** [comment]
Thank you for report this @bryanjhv.  I can reproduce the problem.
It seems my test case was too simple: https://github.com/fastify/fluent-schema/blob/d3e56231170728e89cabc4be12ca3bdf736c3c38/src/BaseSchema.test.js#L125-L132
I will have a look

**Comment by aboutlo** [comment]
fixed in v0.6.2

**Comment by bryanjhv** [comment]
Thanks! It is fixed now.
